test/test_serialization.py

```diff
@@ -0,0 +1,314 @@
+
+        for dpipe, dp_args, dp_kwargs in picklable_datapipes:
+            try:
+                # Creating input (usually a DataPipe) for the specific dpipe being tested
+                if dpipe in custom_input:
+                    input_for_dp = custom_input[dpipe]
+                else:
+                    input_for_dp = IterableWrapper(range(10))
+
+                if dpipe in dp_skip_comparison:  # Mke sure they are picklable and loadable (no value comparison)
+                    datapipe = dpipe(input_for_dp, *dp_args, **dp_kwargs)  # type: ignore[call-arg]
+                    serialized_dp = pickle.dumps(datapipe)
+                    _ = pickle.loads(serialized_dp)
+                elif dpipe in dp_compare_children:  # DataPipes that have children
+                    dp1, dp2 = dpipe(input_for_dp, *dp_args, **dp_kwargs)  # type: ignore[call-arg]
+                    self._serialization_test_for_dp_with_children(dp1, dp2)
+                else:  # Single DataPipe that requires comparison
+                    datapipe = dpipe(input_for_dp, *dp_args, **dp_kwargs)  # type: ignore[call-arg]
+                    is_dataframe = issubclass(dpipe, (iterdp.DataFrameMaker, iterdp.ParquetDataFrameLoader))
+                    self._serialization_test_for_single_dp(datapipe, is_dataframe=is_dataframe)
+            except Exception as e:
+                print(f"{dpipe} is failing.")
+                raise e
+
+    def test_serializable_with_dill(self):
+        """Only for DataPipes that take in a function as argument"""
+        input_dp = IterableWrapper(range(10))
+        ref_idp = IterableWrapper(range(10))
+        ref_mdp = SequenceWrapper(range(10))
+
+        unpicklable_datapipes: List = [
+            (iterdp.FlatMapper, (lambda x: [x, x],), {}),
+            (iterdp.IterKeyZipper, (ref_idp, lambda x: x, None, True, 100), {}),
+            (iterdp.MapKeyZipper, (ref_mdp, lambda x: x), {}),
+            (iterdp.OnDiskCacheHolder, (lambda x: x,), {}),
+            (iterdp.ParagraphAggregator, (lambda x: x,), {}),
+        ]
+        # Skipping value comparison for these DataPipes
+        dp_skip_comparison = {iterdp.OnDiskCacheHolder, iterdp.ParagraphAggregator}
+        for dpipe, dp_args, dp_kwargs in unpicklable_datapipes:
+            if DILL_AVAILABLE:
+                try:
+                    if dpipe in dp_skip_comparison:  # Make sure they are picklable/loadable (no value comparison)
+                        datapipe = dpipe(input_dp, *dp_args, **dp_kwargs)  # type: ignore[call-arg]
+                        serialized_dp = dill.dumps(datapipe)
+                        _ = dill.loads(serialized_dp)
+                    else:
+                        datapipe = dpipe(input_dp, *dp_args, **dp_kwargs)  # type: ignore[call-arg]
+                        self._serialization_test_for_single_dp(datapipe)
+                except Exception as e:
+                    print(f"{dpipe} is failing.")
+                    raise e
```

Comment:
Since we iterating through different DataPipes in a for loop, I added this part so that we actually know which DataPipe is failing. Open to better ideas.